tests/Fixer/ControlStructure/EmptyLoopConditionFixerTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ControlStructure;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ControlStructure\EmptyLoopConditionFixer
+ */
+final class EmptyLoopConditionFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFixConfig(string $expected, ?string $input = null, ?array $config = null): void
+    {
+        if (null !== $config) {
+            $this->fixer->configure($config);
+        }
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases(): \Generator
+    {
+        yield 'from `for` to `while`' => [
+            '<?php while(true){ if(foo()) {break;}}',
+            '<?php for(;;){ if(foo()) {break;}}',
+        ];
+
+        yield 'from `do while` to `while`' => [
+            '<?php while(true){ if(foo()) {break;}}',
+            '<?php do{ if(foo()) {break;}}while(true);',
+        ];
+
+        yield 'from `while` to `for`' => [
+            '<?php for(;;){ if(foo()) {break;}}',
+            '<?php while(true){ if(foo()) {break;}}',
+            ['style' => 'for'],
+        ];
+
+        yield 'from `do while` to `for`' => [
+            '<?php for(;;){ if(foo()) {break;}}',
+            '<?php do{ if(foo()) {break;}}while(true);',
+            ['style' => 'for'],
+        ];
+
+        yield 'multiple `do while` to `while`' => [
+            '<?php while(true){}while(true){}while(true){}while(true){}while(true){}while(true){}',
+            '<?php do{}while(true);do{}while(true);do{}while(true);do{}while(true);do{}while(true);do{}while(true);',
+        ];
+
+        yield 'multiple nested `do while` to `while`' => [
+            '<?php while(true){while(true){while(true){while(true){while(true){while(true){while(true){}}}}}}}',
+            '<?php do{do{do{do{do{do{do{}while(true);}while(true);}while(true);}while(true);}while(true);}while(true);}while(true);',
+        ];
+
+        // comment cases
+
+        yield 'comment inside empty `for` condition' => [
+            '<?php while(true)/* 1 *//* 2 */{}',
+            '<?php for(/* 1 */;;/* 2 */){}',
+        ];
+
+        yield 'comment following empty `for` condition' => [
+            '<?php for(;;)/* 3 */{}',
+            '<?php while(true/* 3 */){}',
+            ['style' => 'for'],
+        ];
+
+        // space cases
+        yield 'lot of space' => [
+            '<?php while(true){ foo3(); }              ',
+            '<?php do{ foo3(); } while(true)            ; ',
+        ];
+
+        yield [
```

Comment:
this change shows that meaning of code is executed.
`while` loop may never execute the body of control statement, while `do-while` executes it at least once

https://3v4l.org/L9ILN#v8.0.9

Reply:
I don't think so, only empty conditions are fixed, not variables, just true

Reply:
Ican add your example as test, showing it won't get touched

Reply:
would love to add it. even if current code is smart enough to handle it, let's avoid someone introducing this issue while refactoring

Reply:
thanks, added :+1: